### ISSUE TITLE
Use accessibility-based locators in e2e tests

### ISSUE THREAD

**Issue by cquinn540** [activist-org/activist#1048]
### Terms

- [X] I have searched [open and closed feature requests](https://github.com/activist-org/activist/issues?q=is%3Aissue+label%3Afeature)
- [X] I agree to follow activist's [Code of Conduct](.github/CODE_OF_CONDUCT.md)

### Description

This should be done after #1047

Currently the locators in the e2e tests use mostly ids and class names to locate elements.  This should be changed to use accessibility locators as that is how our flesh-and-blood human users will locate elements in real life.

Some of the components themselves will need to be made more accessible for this to work, so this should be done in pieces with multiple PRs to keep the changes from getting too large.

### Contribution

I plan to work on this

**Comment by aasimsyed** [comment]
I can work on this.

**Comment by andrewtavis** [comment]
Thanks, @aasimsyed! :)

**Comment by cquinn540** [comment]
@andrewtavis  Looks like I missed that we got two pull requests for this issue:

- https://github.com/activist-org/activist/pull/1122 from @aasimsyed 
- https://github.com/activist-org/activist/pull/1117 from @Abhi-Bohora 

I think want we want is a combination of both approaches:

I like the function signature from 1122:
```typescript
export function getLocaleData(locale?: LOCALE_CODE): LocaleFile
```
since that handles multiple languages.

On the other hand, I think it's probably simpler to just use the module system to read json files like in 1117. Imported json modules use the commonjs cache so we don't have to write our own.  Though @Abhi-Bohora you should check and see if your import works in playwright if you are running the latest Node v20.

My understanding is that in Node v20+ you need to use [import attributes](https://nodejs.org/docs/latest-v20.x/api/esm.html#import-attributes) to import json. We should also probably document that we use Node v20 if it we haven't already.

Maybe it makes sense for @Abhi-Bohora to modify @aasimsyed 's branch so they both get commits in for working on this?

**Comment by andrewtavis** [comment]
We can also bring in @aasimsyed's and then @Abhi-Bohora can merge main into #1117. Not sure on the write access of @Abhi-Bohora to #1122 🤔 Open to whatever works for the contributors though :)

**Comment by Abhi-Bohora** [comment]
Hello 👋 @andrewtavis  @cquinn540  @aasimsyed  what do you think of this way of combining both approach ? 🤔
```ts
// SPDX-License-Identifier: AGPL-3.0-or-later
import { LOCALE_CODE } from "~/locales";
import enUS from "../../frontend/i18n/en-US.json" assert { type: "json" };
import de from "../../frontend/i18n/de.json" assert { type: "json" };
import es from "../../frontend/i18n/es.json" assert { type: "json" };
import fr from "../../frontend/i18n/fr.json" assert { type: "json" };
import pt from "../../frontend/i18n/pt.json" assert { type: "json" };

type LocaleFile = Record<string, string>;

const localeFiles: Record<LOCALE_CODE, LocaleFile> = {
  [LOCALE_CODE.ENGLISH]: enUS,
  [LOCALE_CODE.GERMAN]: de,
  [LOCALE_CODE.SPANISH]: es,
  [LOCALE_CODE.FRENCH]: fr,
  [LOCALE_CODE.PORTUGUESE]: pt,
};

export function getLocaleText(locale?: LOCALE_CODE): LocaleFile {
  return localeFiles[locale || LOCALE_CODE.ENGLISH];
}

export const getI18nString = (key: string) => {
  return getLocaleData(LOCALE_CODE.ENGLISH)[key];
};

```

**Comment by cquinn540** [comment]
Yes, that looks good to me!

**Comment by andrewtavis** [comment]
To me as well, @Abhi-Bohora! Please feel free to update your PR and we'll then likely merge in @aasimsyed's and then yours after to finalize the changes :)
